### DISCUSSION TITLE
Panel plugin for Mouser to create a shopping cart from a PO

### DISCUSSION THREAD

**Discussion by SergeoLacruz** [inventree/InvenTree#4694] [Show and tell]
Hi,
I created a plugin that allows to transfer an InvenTree PO into a mouser shopping cart. The plugin is ready to be used. Anyhow there are things I'd like to impove in the near future. Some Ideas are at the end of the docs.  Has anyone an idea how to come from the purchaseorderdetailview to the supplier in an easy way?
If you like have a look and send feedback. the code is here:

https://github.com/SergeoLacruz/inventree-supplier-panel

Michael

**Comment by SchrodingersGat** [comment]
@SergeoLacruz I really like what you have done here! In future I would like to provide some of this functionality in a generic fashion - there is some stuff in the works here.

It should be quite straight-forward to determine the supplier ID from the PurchaseOrderDetail page:

```python
SETTINGS = {
    ...,
    {
        'MOUSER_PK': {
            'name': 'Mouser Supplier ID',
            'description': 'Primary key of the Mouser supplier',
            'model': 'company.company',
        }
    }
}

def get_custom_panels(self, view, request):
    panels = []

    # Extract ID for Mouser supplier
    mouser_pk = int(self.get_setting('MOUSER_PK'))

    # Only display this panel for orders from "Mouser"
    if isinstance(view, PurchaseOrderDetail):
        order = instance.get_object()
        if order.supplier.pk == mouser_pk:
            panels.append({
                'title': 'Mouser Actions',
                'icon': 'fa-user',
                'content_template': 'supplier_panel/mouser.html', 
            })

    return panels
```

**Reply by wolflu05** [reply]
Where does instance come from?

**Reply by SchrodingersGat** [reply]
Oops that should be `view.get_object`

**Comment by eeintech** [comment]
Very nice plugin, keep up the great work!!

**Comment by SergeoLacruz** [comment]
Thanks for the hint. That's really easy. I pushed an update :-)

**Comment by wolflu05** [comment]
This is a really cool idea for a plugin. But I'm not sure what the scope of your plugin is. Are you planning to extend this to other suppliers as well as your generic title indicates? (Btw, your description contains a spelling mistake - chopping cart)

**Comment by SergeoLacruz** [comment]
Fixwd. Thanks.
Yes. As Digikey managed to approve my account after six weeks of discussions about a too long email address...
This will be the next. I have seen that something already exists. 
But first we need again something that adds supplier parts more or less automatically. 
Will take time.

**Reply by wolflu05** [reply]
You're E-Mail address is too long 😃? 

> I have seen that something already exist

What do you mean by this?

Due to adding the supplier parts, #3791 would be handy. I'm also waiting for it until I add all of my components 😃 - I use reichelt as a supplier for most of my components, but unfortunately they don't have a real official api.

**Reply by wolflu05** [reply]
By the way, I saw you sometimes reply using a new comment. Gh discussions can have comments and each comment can have sub comments. If you specifically reply to a comment, it's easier to see what belongs together.

**Reply by matmair** [reply]
Digikey can be... interesting sometimes when needing tech support. Their purchasing dep. is top-notch though.

**Comment by matmair** [comment]
Looks very nice Michael!

**Comment by SergeoLacruz** [comment]
I did some additions to the plugin. The error handling of Mouser requests is improved. You can see now if there are problems with a spesicif part in our PO e.g. SKU not available. 
There is also Digikey integration now including their token based authorization system. This is still experimental and has not been fully tested. As Digikey does not have a shopping cart API I use the MyLists API which is more complex . There ares still issues with that.

**Comment by SergeoLacruz** [comment]
I finished the Digikey integration and added the doku for it. Setting it up is a bit more complicated. It works at least in my environment. If anybody likes to try it, you are welcome.

**Comment by Hauzman** [comment]
@SergeoLacruz hi can you tell me where I can put the Mouser API bebouase i cannot find wher to put it 
And for ex. in the purchase order It is not apearing the mouser action tab.

**Reply by SergeoLacruz** [reply]
Mouser is easy. Just add your search and cart key into the appropiate fields and written in the docs.

**Reply by Hauzman** [reply]
I will try it again to re instal it because i cannot find the configuration where to add the API . I have just this only in the plug in list :

<img width="2383" height="1175" alt="Screenshot 2025-11-13 at 21 56 26" src="https://github.com/user-attachments/assets/c8a1a7a2-a839-4a96-9ef2-b47a8796dc62" />
<img width="1255" height="678" alt="Screenshot 2025-11-13 at 21 59 59" src="https://github.com/user-attachments/assets/74c97a4b-ebab-4c13-a513-205db2274a7b" />

<img width="2479" height="699" alt="Screenshot 2025-11-13 at 22 00 49" src="https://github.com/user-attachments/assets/d1005461-f142-4db1-92b4-b7d2702987db" />

**Reply by SergeoLacruz** [reply]
Sorry for ansering this late. 
From you screenshot I see that you are using the new platform UI, probably Inventree vesion 1.something.
The plugin was developed for the old CUI and does not support PUI. Sorry.
